src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -317,123 +324,44 @@ protected override IEnumerable<string> GetSimpleReferences (Type type)
 			};
 		}
 
+		[DllImport (AndroidRuntime.InternalDllName, CallingConvention = CallingConvention.Cdecl)]
+		static extern string? _monodroid_lookup_replacement_type (string jniSimpleReference);
```

Comment:
Unless `_monodroid_lookup_replacement_type()` is allocating memory with the "COM task memory allocator" (iirc `g_malloc()` & `g_free()` on Mono + Unixy platforms), this is Not Good™.   This should probably be `IntPtr`.

Reply:
See also:

* https://www.mono-project.com/docs/advanced/pinvoke/#strings-as-return-values
* https://www.mono-project.com/docs/advanced/pinvoke/#classes-and-structures-as-return-values

Reply:
> Unless `_monodroid_lookup_replacement_type()` is allocating memory with the "COM task memory allocator" (iirc `g_malloc()` & `g_free()` on Mono + Unixy platforms), this is Not Good™. This should probably be `IntPtr`.

Memory is in a read-only string section in ELF